test/extended/machine_config/boot_image_update_aws.go

```diff
@@ -0,0 +1,53 @@
+package machine_config
+
+import (
+	"context"
+	"path/filepath"
+
+	osconfigv1 "github.com/openshift/api/config/v1"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+// This test is [Serial] because it modifies the cluster/machineconfigurations.operator.openshift.io object in each test.
+var _ = g.Describe("[sig-coreos][OCPFeatureGate:ManagedBootImagesAWS][Serial]", func() {
+	defer g.GinkgoRecover()
+	var (
+		MCOMachineConfigurationBaseDir = exutil.FixturePath("testdata", "machine_config", "machineconfigurations")
+		partialMachineSetFixture       = filepath.Join(MCOMachineConfigurationBaseDir, "managedbootimages-partial.yaml")
+		allMachineSetFixture           = filepath.Join(MCOMachineConfigurationBaseDir, "managedbootimages-all.yaml")
+		noneMachineSetFixture          = filepath.Join(MCOMachineConfigurationBaseDir, "managedbootimages-none.yaml")
+		oc                             = exutil.NewCLIWithoutNamespace("machine-config")
+	)
+
+	g.BeforeEach(func(ctx context.Context) {
+		//skip this test if not on GCP platform
+		skipUnlessTargetPlatform(oc, osconfigv1.AWSPlatformType)
+		//skip this test if the cluster is not using MachineAPI
+		skipUnlessFunctionalMachineAPI(oc)
+	})
+
+	g.AfterEach(func() {
+		// Clear out boot image configuration between tests
+		err := oc.Run("apply").Args("-f", noneMachineSetFixture).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.It("Should update boot images only on MachineSets that are opted in [apigroup:machineconfiguration.openshift.io]", func() {
+		PartialMachineSetTest(oc, partialMachineSetFixture)
+	})
+
+	g.It("Should update boot images on all MachineSets when configured [apigroup:machineconfiguration.openshift.io]", func() {
```

Comment:
FWIW this one [failed](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/29099/pull-ci-openshift-origin-master-e2e-aws-ovn-single-node-techpreview-serial/1838305875253530624)

Reply:
This is strange...looking at the logs it seems like the MachineConfiguration object was out of date when the MCO's controller was trying to roll out the boot image update(resulting in the update not taking place). Perhaps some kind of SNO unstability, resulting in slow API requests? 
```
E0923 23:08:14.916835       1 leaderelection.go:340] Failed to update lock optimitically: Put "https://172.30.0.1:443/apis/coordination.k8s.io/v1/namespaces/openshift-machine-config-operator/leases/machine-config-controller": dial tcp 172.30.0.1:443: connect: connection refused, falling back to slow path
E0923 23:08:14.917493       1 leaderelection.go:347] error retrieving resource lock openshift-machine-config-operator/machine-config-controller: Get "https://172.30.0.1:443/apis/coordination.k8s.io/v1/namespaces/openshift-machine-config-operator/leases/machine-config-controller": dial tcp 172.30.0.1:443: connect: connection refused
```
It almost seems like the node was in the middle of an update which doesn't make sense :thinking: